zetta_utils/mazepa_layer_processing/alignment/aced_relaxation_flow.py

```diff
@@ -67,16 +69,19 @@ def __call__(
         max_dist: int,
     ):
         idx_padded = idx.padded(self.crop_pad)
-        tissue_mask_data = tissue_mask[idx_padded]
-        if (tissue_mask_data != 0).sum() > 0:
-            result = alignment.aced_relaxation.get_aced_match_offsets(
-                tissue_mask=tissue_mask_data,
-                misalignment_masks={k: v[idx_padded] for k, v in misalignment_masks.items()},
-                pairwise_fields={k: v[idx_padded] for k, v in pairwise_fields.items()},
-                pairwise_fields_inv={k: v[idx_padded] for k, v in pairwise_fields_inv.items()},
-                max_dist=max_dist,
-            )
+        with semaphore("read"):
+            tissue_mask_data = tissue_mask[idx_padded]
+        with semaphore("cuda"):
+            if (tissue_mask_data != 0).sum() > 0:
+                result = alignment.aced_relaxation.get_aced_match_offsets(
+                    tissue_mask=tissue_mask_data,
+                    misalignment_masks={k: v[idx_padded] for k, v in misalignment_masks.items()},
+                    pairwise_fields={k: v[idx_padded] for k, v in pairwise_fields.items()},
+                    pairwise_fields_inv={k: v[idx_padded] for k, v in pairwise_fields_inv.items()},
+                    max_dist=max_dist,
+                )
             result = {k: tensor_ops.crop(v, self.crop_pad) for k, v in result.items()}
```

Comment:
I think this still might be missing an indent. Isnt result not necessarily defined here?